src/test/StdCheats.t.sol

```diff
@@ -138,6 +138,11 @@ contract StdCheatsTest is Test {
         assertEq(string(getCode(deployed)), string(getCode(address(this))));
     }
 
+    function testDeployCodeFail() public {
+        vm.expectRevert(bytes("StdCheats deployCode(string memory): Deployment failed"));
```

Comment:
@ZeroEkkusu this `expectRevert` is not working as expected, even though error is being thrown.

Reply:
This is because when `expectRevert` is used before `deployCode`, the value will not get assigned to `addr` in `deployCode`, regardless of the success:

https://github.com/foundry-rs/forge-std/blob/4d36e3f7e2168c8155c641eb0f80e85cd584bd1c/src/Test.sol#L163

Would you please take a look at this, @mds1?

Reply:
Sorry not sure I'm following—don't we want `addr` to be the zero address, so the require statement is hit and it reverts?

Or is the issue that expectRevert isn't working here because we're not actually making a call? If so, you can try making it an external call with `this.deployCode` or using a try/catch and verifying the revert reason

Reply:
Thanks @mds1, `this.deployCode` solves the issue.

@dsam82, don't forget to update the `bound` tests to use the new messages too!

Reply:
Wow! this works. Can you make my dumbass understand why? 😅

Reply:
`this.deployCode` works because it makes a `CALL`, whereas `deployCode` is making a JUMP which `expectRevert` does not hook into